.github/workflows/flutter.yaml

```
@@ -20,23 +20,46 @@ jobs:
         with:
           channel: 'stable'
       
+      - name: Check flutter sdk version
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: flutter --version
+      
       - name: Get dependencies
         if: steps.check_files.outputs.files_exists == 'true'
         run: flutter pub get
-
-      - name: Run static code analysis
+      
+      - name: Setup Dart Code Metrics
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: invertase/github-action-dart-analyzer@v1
-        with:
-          fatal-infos: true
+        run: dart pub global activate dart_code_metrics
 
       - name: Dart Code Metrics
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: dart-code-checker/dart-code-metrics-action@v2
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          fatal_warnings: true
-          fatal_style: true
+        run: |
+          dirs_to_analyse=""
+          if [ -d lib ]
+          then
+            dirs_to_analyse+=" lib"
+          fi
```

Nit: can be shortened:
```suggestion
          if [ -d lib ]; then dirs_to_analyse+=" lib" ; fi
```